hadoop-common-project/hadoop-common/src/main/native/src/org/apache/hadoop/io/nativeio/NativeIO.c

```diff
@@ -1341,6 +1350,68 @@ Java_org_apache_hadoop_io_nativeio_NativeIO_00024Windows_extendWorkingSetSize(
 #endif
 }
 
+jint
+fdval(JNIEnv *env, jobject fdo)
+{
+    return (*env)->GetIntField(env, fdo, fd_fdID);
+}
+
+jlong
+handleval(JNIEnv *env, jobject fdo)
+{
+    return (*env)->GetLongField(env, fdo, handle_fdID);
+}
+
+JNIEXPORT jlong JNICALL
+Java_org_apache_hadoop_io_nativeio_NativeIO_00024Windows_transmitFile(
+  JNIEnv *env, jobject jObject, jobject srcFD, jlong position, jlong count,
+  jobject dstFD)
+{
+#ifdef UNIX
+  THROW(env, "java/io/IOException",
+    "The function transmitFile is not supported on Unix");
+#endif
+
+#ifdef WINDOWS
```

Comment:
#elseif WINDOWS

Reply:
i was following the way this was done just above for Windows_extendWorkingSetSize